### PR TITLE
Converted jquery.dragster.js from jquery plugin to vanilla js

### PR DESCRIPTION
**Pull request by niklabh**
#### Summary

Converted jquery.dragster.js from jquery plugin to vanilla js

#### Ticket Link

Fixes: https://github.com/mattermost/mattermost-server/issues/11429

### CODE REVIEW

File: utils/dragster.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            event.preventDefault();
+        });
+
+        node.addEventListener('dragster:enter', settings.enter);
+        node.addEventListener('dragster:leave', settings.leave);
+        node.addEventListener('dragster:over', settings.over);
+        node.addEventListener('dragster:drop', settings.drop);
+    });
+}
+
```
[the remark this answers is not in the document]
Comment (niklabh) [reply utils/dragster.js]: Please suggest a way to unit test. Any help will be appreciated.

**Comment by niklabh** [comment]
> Also, this fails on existing test
> Screen Shot 2019-09-12 at 4 51 17 PM

Tests passing on local.

File: components/file_upload/file_upload.jsx [hunk 2]
```diff
@@ -397,55 +396,55 @@ export default class FileUpload extends PureComponent {
...
                     dragTimeout.cancel();
                 },
                 over() {
                     dragTimeout.fireAfter(OVERLAY_TIMEOUT);
                 },
-                drop(dragsterEvent, e) {
-                    if (!overlay.hasClass('hidden')) {
-                        overlay.addClass('hidden');
+                drop(e) {
+                    if (!overlay.classList.contains('hidden')) {
```
[the remark this answers is not in the document]
Comment (niklabh) [reply components/file_upload/file_upload.jsx]: done

File: components/file_upload/file_upload.jsx [hunk 3]
```diff
@@ -397,55 +396,55 @@ export default class FileUpload extends PureComponent {
...
                     }
                 },
-                leave(dragsterEvent, e) {
-                    var files = e.originalEvent.dataTransfer;
+                leave(e) {
+                    var files = e.detail.dataTransfer;
 
-                    if (isFileTransfer(files) && !overlay.hasClass('hidden')) {
-                        overlay.addClass('hidden');
+                    if (isFileTransfer(files) && !overlay.classList.contains('hidden')) {
```
[the remark this answers is not in the document]
Comment (niklabh) [reply components/file_upload/file_upload.jsx]: done

File: components/file_upload/file_upload.jsx [hunk 4]
```diff
@@ -397,55 +396,55 @@ export default class FileUpload extends PureComponent {
     registerDragEvents = (containerSelector, overlaySelector) => {
         const self = this;
 
-        const overlay = $(overlaySelector);
+        const overlay = document.querySelector(overlaySelector);
 
         const dragTimeout = new DelayedAction(() => {
-            if (!overlay.hasClass('hidden')) {
-                overlay.addClass('hidden');
+            if (!overlay.classList.contains('hidden')) {
```
[the remark this answers is not in the document]
Comment (niklabh) [reply components/file_upload/file_upload.jsx]: done

**Comment by niklabh** [comment]
Removed target.off jquery function in last commit. Using removeEventListener from inside the lib using js closure.

**Comment by niklabh** [comment]
Doing querySelector instead of querySelectorAll inside the lib now.

**Comment by saturninoabril** [comment]
> Tests passing on local.

I can see it's failing.  Curious, how do you run the test on local?
Screen Shot 2019-09-13 at 9 30 36 PM

File: utils/dragster.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (niklabh) [reply utils/dragster.js]: added tests

**Comment by niklabh** [comment]
> > Tests passing on local.
> 
> I can see it's failing. Curious, how do you run the test on local?
> Screen Shot 2019-09-13 at 9 30 36 PM

Passing now.

**Comment by jespino** [comment]
Merged! Thanks @niklabh! :tada:

**Pull request merged**